File: test/leetcode/matrix/valid-word-square.test.ts
import { validWordSquare } from '../../src/matrix/valid-word-square';

describe('valid word square', () => {
  test('valid word square - test case 1', async () => {
    const words = ['abcd', 'bnrt', 'crmy', 'dtye'];

    expect(validWordSquare(words)).toBe(true);
  });

  test('valid word square - test case 2', async () => {
    const words = ['abcd', 'bnrt', 'crm', 'dt'];

    expect(validWordSquare(words)).toBe(true);
  });
});

File: test/leetcode/matrix/valid_word_square_test.py
from leetcode.matrix.valid_word_square import Solution


soln = Solution()


def test_case_1():
    assert soln.validWordSquare(["abcd", "bnrt", "crmy", "dtye"])


def test_case_2():
    assert soln.validWordSquare(["abcd", "bnrt", "crm", "dt"])
